app/helpers/fhir_helper.rb

```diff
@@ -284,7 +299,10 @@ def history_as_fhir(history)
       ],
       extension: [
         to_string_extension(history.comment, 'comment'),
-        to_string_extension(history.history_type, 'history-type')
+        to_string_extension(history.history_type, 'history-type'),
+        to_string_extension(history.deleted_by, 'deleted-by'),
+        to_string_extension(history.delete_reason, 'delete-reason'),
```

Comment:
Not trying to over-complicate this, but these could be captured with a `contained` `Provenance` as well, similar to with created data. It would still require us to extend `Provenance.reason` in order to properly capture the delete reason, but still it would probably be better in the eyes of the FHIR gods.

Reply:
Done here: https://github.com/SaraAlert/SaraAlert/pull/1016/commits/9e1c5b559b65df44d66e76a813a886da053d61f5. I put the `delete-reason` extension in the general extension array, since it felt a little fishy to put it on `Provenance.reason` and then not actually make an attempt to set any sort of CodeableConcept. Maybe that is just fine and I'm being overly sensitive, and let me know if you think so, but that was just my thinking.

Reply:
I was not clear but by extending `Provenance.reason` I was thinking more like defining our own code system or just using `reason.text`

Reply:
Ok yeah I think this is a viable use of `reason.text`.